WISE-Community/WISE
# Discussion post preview image attachment bug fix

This PR fixes the issue where attaching an image to a discussion new post compose was not showing up in post preview. I also cleaned up surrounding code a little bit.

How to verify:
1. Sign in as a student to a project that has as Discussion component
2. Go to the Discussion component
3. Click the red "Add picture" button
4. Choose an image
5. Click "Close"
6. The image should show up in the post preview. (Images were not showing up before)
7. If you click "Post", the post should be submitted and the image should show up in the submitted post
8. Verify that you can create a new post with only image(s).
9. Verify that you can create a new post with only text.
10. Verify that you can create a new post with image(s) and text.

Fixes #1876.
